mongomock/collection.py

```diff
@@ -1754,15 +1754,41 @@ def _extend_collection(out_collection, field, expression):
                                 unwound_collection[-1], v[1:], field_item)
                     out_collection = unwound_collection
                 elif k == '$project':
-                    filter_list = ['_id']
+                    filter_list = []
+                    method = None
+                    include_id = v.get('_id')
+                    if not v:
+                        method = 'include' if include_id else 'exclude'
```

Comment:
Actually `$project` should raise `pymongo.errors.OperationFailure: Invalid $project :: caused by :: specification must have at least one field` if `v` is empty.

I think the case that was not handled properly below was the case where `v` only includes `_id`. I think the easiest is to move the check at the bottom of the for loop and check whether `method` is set or not yet.

Reply:
Yeah, sorry. Previously I was using `pop` to grab `_id`, and when I did this change I still had that in mind.